### PR TITLE
Deprected module airflow.providers.cncf.kubernetes.pod_launcher_deprecated removed

### PR DESCRIPTION
**Pull request by dirrao**
Deprected module airflow.providers.cncf.kubernetes.pod_launcher_deprecated removed

### CODE REVIEW

**Review by eladkal — CHANGES_REQUESTED** [review]
This is provider breaking change not Airflow 3...
I don't think we have a reason to introduce breaking change of K8s provider right now

File: newsfragments/41830.significant.rst
Comment (jscheffl) [inline newsfragments/41830.significant.rst:1]: Newsfragemnts are only for core, not for providers.

Comment (dirrao) [reply newsfragments/41830.significant.rst:1]: removed

**Comment by dirrao** [comment]
> This is provider breaking change not Airflow 3...
> I don't think we have a reason to introduce breaking change of K8s provider right now

True. We can do whenever we do major provider version changes.

**Comment by vincbeck** [comment]
Closing this one

**Pull request closed without merging**